### PR TITLE
Fix sort sparse vectors

### PR DESCRIPTION
**Pull request by joein**
#439

### CODE REVIEW

File: qdrant_client/local/local_collection.py [hunk 1]
```diff
@@ -401,7 +401,7 @@ def search(
             scores = calculate_context_scores(query_vector, vectors[: len(self.payload)], distance)
         elif isinstance(query_vector, SparseVector):
             # sparse vector query must be sorted by indices for dot product to work with persisted vectors
-            sort(query_vector)
+            query_vector = sort_sparse_vector(query_vector)
```
Comment (agourlay) [inline qdrant_client/local/local_collection.py:404]: That's embarrassing, I guess I thought this was a sort in place :face_with_spiral_eyes:

File: qdrant_client/local/sparse.py [hunk 2]
```diff
@@ -24,18 +28,20 @@ def is_sorted(vector: SparseVector) -> bool:
...
-def sort(vector: SparseVector) -> SparseVector:
+def sort_sparse_vector(vector: SparseVector) -> SparseVector:
     if is_sorted(vector):
         return vector
 
     sorted_indices = np.argsort(vector.indices)
     return SparseVector(
-        indices=vector.indices[sorted_indices],
-        values=vector.values[sorted_indices],
+        indices=[vector.indices[i] for i in sorted_indices],
```
Comment (agourlay) [inline qdrant_client/local/sparse.py:37]: Why is this change necessary?

File: qdrant_client/local/local_collection.py (hunk 1 above)
Comment (joein) [reply qdrant_client/local/local_collection.py:404]: 🙈

File: qdrant_client/local/sparse.py (hunk 2 above)
Comment (joein) [reply qdrant_client/local/sparse.py:37]: `vector.indices` and `vector.values` are built-in python lists
This kind of lists does not support indexing by array of indices

To take elements from an array by several indices, this array should support such an operation (e.g. np.array)

**Review by agourlay — APPROVED** [review]
Thank you!

**Review by generall — APPROVED** [review]
